internal/services/apimanagement/schemaz/api_management.go

```diff
@@ -419,8 +424,18 @@ func FlattenApiManagementOperationParameterExampleContract(input map[string]*api
 			output["description"] = *v.Description
 		}
 
+		// value can be any type, may be a primitive value or an object
+		// https://github.com/Azure/azure-sdk-for-go/blob/main/services/apimanagement/mgmt/2021-08-01/apimanagement/models.go#L10236
 		if v.Value != nil {
-			output["value"] = v.Value.(string)
+			if val, ok := v.Value.(string); ok {
+				output["value"] = val
+			} else {
+				val, err := json.Marshal(v.Value)
+				if err != nil {
+					return nil, fmt.Errorf(" failed to marshal `request/response.representations.examples.value` to json. Error: %+v", err)
+				}
+				output["value"] = val
+			}
```

Comment:
```suggestion
			value := ""
			if val, ok := v.Value.(string); ok {
				value = val
			} else {
				val, err := json.Marshal(v.Value)
				if err != nil {
					return nil, fmt.Errorf("failed to marshal `representations.examples.value` to json: %+v", err)
				}
				value = string(val)
			}
		}
		output["value"] = value
```

Reply:
@stephybun 
Question 1 : I tried to reproduce this error with tests but failed, the type of `value` is `TypeString` and Terraform does not support set the type generic type like `interface`
I reproduce this error in this way:
  1.  Create new  API operation:
      ```hcl
       - resource "azurerm_api_management_api_operation" "example" {
        operation_id        = "crash-reproduce"
        api_name            = data.azurerm_api_management_api.example.name
        api_management_name = data.azurerm_api_management_api.example.api_management_name
        resource_group_name = data.azurerm_api_management_api.example.resource_group_name
        display_name        = "rash API"
        method              = "DELETE"
        url_template        = "/users/{id}/done"
        description         = "This can only be done by the logged in user."
      
        template_parameter {
          name     = "id"
          required = true
          type     = "string"
        }
      
        response {
          status_code = 200
          representation {
            content_type = "application/json"
            example {
              name  = "default"
              value = <<JSON
                    {"value":{"value":{"11":"11","22":"22","33":"33"}}}
                  JSON
            }
      
            example {
              name        = "test-example"
              summary     = "just for test"
              description = "will crash?"
              value       = "{\"value\": {\"11\": \"11\",\"22\": \"22\",\"33\": \"33\"}}"
            }
          }
        }
      }
      ```
  2. Change the response example in portal with JSON object and save
      ```json
      {
                  "value": {
                      "value": {
                          "11": "11",
                          "22": "22",
                          "33": "33"
                      }
                  }
              }
      ```
   3.  Run `terraform plan` , if object value not properly handled, there will an error and provider will crash(v2.90~ v2.99.0).  This patch will handle the object values properly.
  Question 2:   Added.